### PR TITLE
document new way of generating routes

### PR DESCRIPTION
**Pull request by dbu**
https://github.com/symfony-cmf/Routing/releases/tag/2.3.0

### CODE REVIEW

**Comment by dbu** [comment]
@acrobat may i ask you to double-check if i got this right? i did not try out these changes...

File: components/routing/dynamic.rst [hunk 1]
```diff
@@ -193,38 +193,38 @@ a route.
...
 :class:`Symfony\\Component\\Routing\\Generator\\UrlGenerator` (which, in turn,
 implements :class:`Symfony\\Component\\Routing\\Generator\\UrlGeneratorInterface`)
-and - if the name is not already a ``Route`` object - loads the Route from the
-Route provider. It then lets the core logic generate the URL from that ``Route``.
+and asks the route provider to find a route based on the name and parameters. It
+then lets the core logic generate the URL from that ``Route``.
 
 The CMF component also includes the ``ContentAwareGenerator``, which extends
-the ``ProviderBasedGenerator``, that checks if ``$name`` is an object
+the ``ProviderBasedGenerator``, that checks if ``_route_object`` is an object
```
Comment (acrobat) [inline components/routing/dynamic.rst:216]:
```suggestion
the ``ProviderBasedGenerator``, that checks if ``_route_object`` parameter is an object
```

File: components/routing/dynamic.rst [hunk 2]
```diff
@@ -193,38 +193,38 @@ a route.
...
 implementing ``RouteReferrersReadInterface``. If it is, it gets the ``Route``
 from that object. Using the ``ContentAwareGenerator``, you can generate URLs
 for your content in three ways:
 
-* Either pass a ``Route`` object as $name
-* Or pass a ``RouteReferrersInterface`` object that is your content as $name
+* Either pass a ``Route`` object as ``_route_object``
```
Comment (acrobat) [inline components/routing/dynamic.rst:221]:
```suggestion
* Either pass a ``Route`` object as the ``_route_object`` parameter
```

File: components/routing/dynamic.rst [hunk 3]
```diff
@@ -193,38 +193,38 @@ a route.
...
+* Or pass a ``RouteReferrersInterface`` object that is your content as ``_route_object``
```
Comment (acrobat) [inline components/routing/dynamic.rst:222]:
```suggestion
* Or pass a ``RouteReferrersInterface`` object that is your content as the ``_route_object`` parameter
```

**Comment by dbu** [comment]
> Question: Should quote all the '_route_object' parameters in the twig examples?

it seems to not matter whether the key is in quotes or not: https://twigfiddle.com/n1llje - which seems a bit strange to me.

**Comment by acrobat** [comment]
Thanks @dbu 👍

**Pull request merged**